packages/pyo3/src/mediator.rs

```diff
@@ -188,29 +188,36 @@ fn communities_to_clustering(
     network: &LabeledNetwork<String>,
     communities: HashMap<String, usize>,
 ) -> Result<Clustering, PyLeidenError> {
-    // if node not found in internal network, bounce
-    // if max(communities) > len(set(communities)), collapse result
-    // if all nodes are mapped, cool
-    // if not all nodes are mapped, generate integers for each new value
-
-    let mut clustering: Clustering = Clustering::as_self_clusters(network.num_nodes());
-
-    let mut all_communities: HashSet<usize> = HashSet::new();
+    // best effort for using a provided community mapping and keeping generated singleton ids
+    // otherwise
+
+    let mut clustering: Clustering = match communities.values().max() {
+        Some(max_community_id) => {
+            // if we have values, we have the max community id, which means we'll actually create
+            // a cluster mapping where every node gets a singleton cluster starting after the actual
+            // last max cluster.
+            let mut cluster_mapping: Vec<usize> = Vec::with_capacity(network.num_nodes());
+            let start: usize = max_community_id + 1;
+            let end: usize = start + network.num_nodes();
+            cluster_mapping.extend(start..end);
```

Comment:
start..end actually is some syntatic sugar around creating an iterator that goes from start to end and emits that number... and extend exhausts an iterator and appends those values to itself (which in this case, happens to be empty)

Reply:
so I'm assuming that in `start..end`, `end` is exclusive?

Reply:
yes, standard for loop style behaviors (this is actually how you loop in rust, with something like `for i in 0..10`, if you want **inclusive** it's `for i in 0..=10`, which isn't the most aesthetically pleasing, but ...